tests/unittests/CoreText/CTLineTests.mm

```diff
@@ -98,4 +98,153 @@
     // comparing secondaryOffset and offset.
     offset = CTLineGetOffsetForStringIndex(line, 4, &secOffset);
     ASSERT_EQ_MSG(offset, secOffset, "Failed: Wrong offset for given index");
+}
+
+TEST(CTLine, CTLineCreateWithAttributedString) {
+    NSMutableAttributedString* attrString = getAttributedString(@"hello");
+    [attrString addAttribute:NSFontAttributeName value:[UIFont systemFontOfSize:20] range:NSMakeRange(3, 2)];
+    CFAttributedStringRef string = (__bridge CFAttributedStringRef)attrString;
+
+    CTLineRef line = CTLineCreateWithAttributedString(string);
+    EXPECT_NE(line, nil);
+
+    EXPECT_EQ(0, CTLineGetStringRange(line).location);
+    EXPECT_EQ(5, CTLineGetStringRange(line).length);
+    EXPECT_EQ(5, CTLineGetGlyphCount(line));
+    CFArrayRef runs = CTLineGetGlyphRuns(line);
+
+    EXPECT_NE(runs, nil);
+    EXPECT_EQ_MSG(2, CFArrayGetCount(runs), "CTLine should create multiple runs when attributed string has multiple attribute ranges");
+    EXPECT_EQ(3, CTRunGetGlyphCount(static_cast<CTRunRef>(CFArrayGetValueAtIndex(runs, 0))));
+    EXPECT_EQ(2, CTRunGetGlyphCount(static_cast<CTRunRef>(CFArrayGetValueAtIndex(runs, 1))));
+
+    // Glyphs for h, e, and l respectively
+    const CGGlyph* helGlyphs = CTRunGetGlyphsPtr(static_cast<CTRunRef>(CFArrayGetValueAtIndex(runs, 0)));
+    ASSERT_NE(nullptr, helGlyphs);
+
+    std::vector<WORD> characters(4);
+    [@"hel" getCharacters:characters.data()];
+    std::vector<CGGlyph> actualGlyphs(4);
+    UIFontDescriptor* fontDescriptor = [UIFontDescriptor fontDescriptorWithName:@"Times New Roman" size:40];
+    UIFont* font = [UIFont fontWithDescriptor:fontDescriptor size:40];
+    CTFontGetGlyphsForCharacters(static_cast<CTFontRef>(font), characters.data(), actualGlyphs.data(), 3);
```

Comment:
nvm, should've read the function documentation first ^^;;;

---
In reply to: [78471645](https://github.com/Microsoft/WinObjC/pull/989#discussion_r78471645) [](ancestors = 78471645,78467363)